src/playdate/system.nim

```diff
@@ -33,9 +36,41 @@ proc getSecondsSinceEpoch* (this: ptr PlaydateSys): tuple[seconds: uint, millise
 type PDCallbackFunction* = proc(): int {.raises: [].}
 var updateCallback: PDCallbackFunction = nil
 
+proc runCatchingTyped*[T](fun: () -> T, fatal: bool = false, messagePrefix: string=""): T =
```

Comment:
@Nycto 
Is there some magic we can add to get rid of the separate typed and untyped variant using a template, macro, or `auto` typing?

Reply:
You should be able to implement a single version of this by using a template like this:

```
template runCatching*(fatal: bool = false, messagePrefix: string = "", body: typed): auto =
    try:
        body
    except:
        getCurrentException().logException(fatal, messagePrefix)
        default(typeof(body))
```